CMakeLists.txt

```diff
@@ -144,6 +144,7 @@ if (VOLK_CPU_FEATURES)
       FORCE)
     set(BUILD_SHARED_LIBS_SAVED "${BUILD_SHARED_LIBS}")
     set(BUILD_SHARED_LIBS OFF)
+    set(ENABLE_INSTALL ON)
```

Comment:
I suggest to go for `set(ENABLE_INSTALL OFF)` because `cpu_features` should never be installed with VOLK. This causes issue #607 . Since we don't need the `cpu_features` library and don't expose any of its APIs, the `NO install` solution would be preferred.

Reply:
I  did try leaving this off before I submit this PR, but it break's the static CI build. This make's sense since the finial stage link needs the cpu_feature library.

One solution to this would be to ENABLE_INSTALL for cpu_features based off if ENABLE_STATIC_LIBS is set

Reply:
I try to add a bit more context here. I suppose package maintainers, e.g. for Debian, explicitly add patches to make sure `cpu_features` is not installed together with with VOLK.
Furthermore, VOLK should not slip into a maintainer role for a dependency. Even if this implies to update the submodule pointer every once in a while. 
As a consequence, we should try to:
1. hide all dependencies as private and not expose them in our API (if possible)
2. not install anything that is not directly VOLK.

Having written all of this, what would be a reasonable approach here?
It's "good news" that only the static build breaks. Right now, I see that the static test fails because one of the tests fails. I suggest to set `ENABLE_INSTALL OFF` and then we would be able to discuss the situation in more detail.
I would really prefer to not install anything related to `cpu_features`.

Reply:
All good, I've added the change.

It does the right thing when cpu_features is installed as a system package on Fedora distro., the volk shared library dynamical links to the the system cpu_features and it does not build the submodule version

Reply:
I asked in the gnuradio matrix chat. I received this hint from @marcusmueller
https://discourse.cmake.org/t/cmake-error-install-export-projecttargets-includes-target-xxx-which-requires-target-yyy-that-is-not-in-any-export-set/7344/3

Reply:
If I read this discussion correctly, this line:
```cmake
target_link_libraries(volk_static PRIVATE CpuFeatures::cpu_features)
```
though, I haven't had the time to fully understand it yet.